whisper-AI-exec-script/action-transcribe-event.xml

```diff
@@ -0,0 +1,240 @@
+	          <label for="transEn">Add automatic transcription to english</label>
+          </li>
+        <!-- Place to keep the layoutType state --> 
+          <!-- Place to keep the layoutType state --> 
+          <input type="hidden" id="layoutType" class="configField" value="NONE" />
+      </fieldset>
+
+      <script>
+          document.getElementById("mTiny").disabled = true;
+          document.getElementById("mSmall").disabled = true;
+          document.getElementById("mLarge").disabled = true;
+
+
+      </script>
+      
+     </div>
+
+  ]]>
+
+  </configuration_panel>
+  
+  <operations>
+  <state-mappings>
+    <state-mapping state="running">EVENTS.EVENTS.STATE_MAPPING.PUBLISHING</state-mapping>
+    <state-mapping state="failing">EVENTS.EVENTS.STATE_MAPPING.PUBLISHING</state-mapping>
+  </state-mappings>
+
+
+    <!-- Set translation parameters-->
+
+    <operation
+      id="conditional-config"
+      exception-handler-workflow="partial-error"
+      description="Setting transcription variables">
+      <configurations>
+        <configuration key="configuration-name">transcription-model</configuration>
+        <configuration key="condition-1">${mTiny}</configuration>
+        <configuration key="value-1">{{ whisper_server }}:9000</configuration>
+        <configuration key="condition-2">${modelBase}</configuration>
+        <configuration key="value-2">{{ whisper_server }}:9001</configuration>
+        <configuration key="condition-3">${mSmall}</configuration>
+        <configuration key="value-3">{{ whisper_server }}:9002</configuration>
+        <configuration key="condition-4">${mMedium}</configuration>
+        <configuration key="value-4">{{ whisper_server }}:9003</configuration>
+        <configuration key="condition-5">${mLarge}</configuration>
+        <configuration key="value-5">{{ whisper_server }}:9004</configuration>
+        <configuration key="no-match">no-model</configuration>
+ 
+      </configurations>
+    </operation>
+
+    <!-- Analyze media assets -->
+
+    <operation
+      id="analyze-tracks"
+      exception-handler-workflow="partial-error"
+      description="|__   Analyze tracks in media package and set control variables">
+      <configurations>
+        <configuration key="source-flavor">*/source</configuration>
+      </configurations>
+    </operation>
+
+    <!-- Delete old subtitles --> 
+
+     <operation
+      id="tag"
+      exception-handler-workflow="partial-error"
+      fail-on-error="false"
+      description="Remove old archived subtitles ">
+      <configurations>
+        <configuration key="source-flavors">captions/vtt+de, captions/vtt+en</configuration>
+        <configuration key="target-tags">-archive,-engage-download</configuration>
+      </configurations>
+    </operation>
+
+    <!-- Generate the subtitles - only if Presenter is available --> 
+
+    <operation
+      id="execute-once"
+      if= "(NOT ${presentation_source_video}) AND ${presenter_source_video}"
```

Comment:
Not necessary, as a helper script is supposed that your media has audio

Reply:
The meaning of [`_audio` here is](https://docs.opencast.org/develop/admin/#workflowoperationhandlers/analyze-tracks-woh/#description) `Track contains at least one audio stream`. This can be an only audio track OR a track with video and audio.

Reply:
Thanks, I didn't understood your review plus I didn't remember that you can do that with analyze-tracks 😜 

I will change it asap